tests/benchmarks/fio_benchmark_test.py

```diff
@@ -153,5 +154,62 @@ def TestZeroMinutes(self):
     self.assertEquals(proc.call_count, 0)
 
 
+class TestFioTargetModeFlag(unittest.TestCase):
+  def doTargetModeTest(self, mode, fill_device=None, against_device=None,
+                       format_disk=None):
+    with mock.patch(fio_benchmark.__name__ + '.FillDevice') as fill_device, \
+            mock.patch(fio_benchmark.__name__ +
+                       '.GetOrGenerateJobFileString') as get_job_string, \
+            mock.patch('__builtin__.open'), \
+            mock.patch(vm_util.__name__ + '.GetTempDir'), \
+            mock.patch(fio_benchmark.__name__ + '.FLAGS') as fio_flags:
+      fio_flags.fio_target_mode = 'against_file_with_fill'
+      fio_flags.fio_run_for_minutes = 0
+      benchmark_spec = mock.MagicMock()
+      fio_benchmark.Prepare(benchmark_spec)
+      fio_benchmark.Run(benchmark_spec)
+
+      if fill_device is True:
+        self.assertEquals(fill_device.call_count, 1)
+      elif fill_device is False:
+        self.assertEquals(fill_device.call_count, 0)
+      # get_job_string.call_args[0][2] is a boolean saying whether or
+      # not we are testing against a device.
+      against_device_arg = get_job_string.call_args[0][2]
+      if against_device is True:
+        self.assertEquals(against_device_arg, True)
+      elif against_device is False:
+        self.assertEquals(against_device_arg, False)
+
+      if format_disk is True:
+        self.assertEquals(benchmark_spec.vms[0].FormatDisk.call_count, 1)
+      elif format_disk is False:
+        self.assertEquals(benchmark_spec.vms[0].FormatDisk.call_count, 0)
+
+    def testAgainstFileWithFill(self):
```

Comment:
I'd expect most of the test cases below to have failed with your current code, that is, the error in:

line 391: "if FLAGS.fio_target_mode == AGAINST_DEVICE_WITH_FILL_MODE: then format disk", 

should have failed 3 of the cases here.

If you see these are all passing even with your current bug, then it's actually a more serious problem, something is wrong with the tests, please dig deeper.